corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -300,38 +340,15 @@ def get_doc(self, doc_id):
     def doc_to_json(self, doc):
         return doc.to_json(include_attachments=self.include_attachments)
 
-    def get_docs(self, from_db, startkey, last_doc_pk=None, limit=500):
-        received_on_since = startkey or datetime.min
-        last_id = last_doc_pk or -1
-
-        domain_clause = "form_table.domain = %s AND" if self.domain else ""
-        values = [received_on_since, last_id, limit]
-        if self.domain:
-            values = [self.domain] + values
-
-        # using raw query to avoid having to expand the tuple comparison
-        results = XFormInstanceSQL.objects.using(from_db).raw(
-            """SELECT * FROM {table} as form_table
-        WHERE {domain_clause}
-        state & {deleted_state} = 0 AND
-        (form_table.received_on, form_table.id) > (%s, %s)
-        ORDER BY form_table.received_on, form_table.id
-        LIMIT %s;""".format(
-                table=XFormInstanceSQL._meta.db_table,
-                domain_clause=domain_clause,
-                deleted_state=XFormInstanceSQL.DELETED
-            ),
-            values
-        )
-        return list(results)
-
-    @property
-    def count_query(self):
-        # deletion clause left out on purpose since it throws off the count estimate
+    def extra_filters(self, for_count=False):
+        filters = []
+        if not for_count:
+            # don't inlucde in count query since the query planner can't account for it
+            # hack for django: 'state & DELETED = 0' so 'state = state + state & DELETED'
+            filters.append(Q(state=F('state').bitand(XFormInstanceSQL.DELETED) + F('state')))
```

Comment:
Does this just mean state != DELETED ? in that case would `~Q(state=DELETED)` be more obvious?

Reply:
Unfortunately not: https://github.com/dimagi/commcare-hq/blob/6ae2006b2a823c4733054de0fc2de17107f26617/corehq/form_processor/models.py#L283-L286

Reply:
Ah that makes sense. Thanks